Runtime/Native/Windows/NativeClient.cs

```diff
@@ -0,0 +1,486 @@
+                ? new List<string>()
+                : new List<string>(clientAttachments);
+
+            // make sure - when user close game in the middle of sending data, the library won't have a chance to clean up temporary breadcurmb
+            // file. Becuase of that we prefer to always check if we need to clean something that left in the previous application session
+
+            string breadcrumbsCopyName = string.Format("{0}-1", BacktraceStorageLogManager.BreadcrumbLogFilePrefix);
+            string breadcrumbCopyPath = Path.Combine(databasePath, breadcrumbsCopyName);
+            if (File.Exists(breadcrumbCopyPath))
+            {
+                File.Delete(breadcrumbCopyPath);
+            }
+
+
+            // determine if handler should create a copy of a breadcrumb file 
+            // on the application startup. This check also prevents a situation when
+            // algorithm will try to copy a breacrumb file when a breadcrumbs file doesn't exist
+            // Client prefers to make a copy of a breadcrumb file in the database directory. Otherwise, if database
+            // for any reason in new session is not available, algorithm shouldn't make a copy. 
+            bool requireBreadcrumbsCopy = string.IsNullOrEmpty(breadcrumbPath) || string.IsNullOrEmpty(databasePath) ? false : true;
+            bool copiedFile = false;
+
+            var crashDirs = Directory.GetDirectories(nativeCrashesDir);
+            foreach (var crashDir in crashDirs)
+            {
+                var crashDirFullPath = Path.Combine(nativeCrashesDir, crashDir);
+                var crashFiles = Directory.GetFiles(crashDirFullPath);
+
+                var alreadyUploaded = crashFiles.Any(n => n.EndsWith("backtrace.json"));
+                if (alreadyUploaded)
+                {
+                    continue;
+                }
+                var minidumpPath = crashFiles.FirstOrDefault(n => n.EndsWith("crash.dmp"));
+                if (string.IsNullOrEmpty(minidumpPath))
+                {
+                    continue;
+                }
+                if (requireBreadcrumbsCopy)
+                {
+                    try
+                    {
+                        File.Copy(breadcrumbPath, breadcrumbCopyPath);
+                        attachments.Add(breadcrumbCopyPath);
+                        copiedFile = true;
+                    }
+                    catch (Exception e)
+                    {
+                        Debug.LogWarning(string.Format("Cannot make a copy of the breadcrumb file in the database directory. Reason: {0}", e.Message));
+                    }
+                    finally
+                    {
+                        requireBreadcrumbsCopy = false;
+                    }
+                }
+                var dumpAttachment = crashFiles.Concat(attachments).Where(n => n != minidumpPath).ToList();
+                IDictionary<string, string> attributes = GetScopedAttributes();
```

Comment:
Should we cache the `GetScopedAttributes` too like we do with Breadcrumbs file now?

Reply:
Yes - we should. Thanks for pointing this out!